silkworm/db/chain/remote_chain_storage.hpp

```diff
@@ -28,14 +28,18 @@ namespace silkworm::db::chain {
 
 using BlockProvider = std::function<Task<bool>(BlockNum, HashAsSpan, bool, Block&)>;
 using BlockNumberFromTxnHashProvider = std::function<Task<BlockNum>(HashAsSpan)>;
+using BlockNumberFromBlockHashProvider = std::function<Task<BlockNum>(HashAsSpan)>;
+using BlockHashFromBlockNumberProvider = std::function<Task<evmc::bytes32>(BlockNum)>;
 
 //! RemoteChainStorage must be used when blockchain data is remote with respect to the running component, i.e. it is
 //! in remote database (accessed via gRPC KV I/F) or remote snapshot files (accessed via gRPC ETHBACKEND I/F)
 class RemoteChainStorage : public ChainStorage {
   public:
```

Comment:
```suggestion
  public:
    struct Providers {
        BlockProvider block;
        BlockNumberFromTxnHashProvider block_number_from_txn_hash;
        BlockNumberFromBlockHashProvider block_number_from_hash;
        BlockHashFromBlockNumberProvider canonical_block_hash_from_number;
    };
```

Reply:
I would define a container for the four providers in order to reduce the number of parameters passed in many places.
Next PR is fine

Reply:
ok